core/src/commonMain/kotlin/org/jetbrains/litmuskt/ThreadlikeRunner.kt

```diff
@@ -5,17 +5,22 @@ abstract class ThreadlikeRunner : LitmusRunner() {
     protected abstract fun threadlikeProducer(): Threadlike
 
     private fun <S : Any> threadFunction(threadContext: ThreadContext<S>) = with(threadContext) {
+        val testFunction = test.threadFunctions[threadIndex]
         for (i in states.indices) {
             if (i % syncPeriod == 0) barrier.await()
-            states[i].testThreadFunction()
+            states[i].testFunction()
         }
+        rangeResult = calcStats(states.view(resultCalcRange).asIterable(), test.outcomeSpec, test.outcomeFinalizer)
```

Comment:
Could you please add a comment here explaining that as an optimization, each thread takes a sublist of all the results, and calculates the statistics for this sublist in parallel to other threads.